### DISCUSSION TITLE
view factor for rear side

### DISCUSSION THREAD

**Discussion by 115522** [pvlib/pvlib-python#1670] [Q&A] [answered]
Hello pvlib_python community !

I have a question about can I calculate view factor for rear side in bifacial system using pvlib 
In fact , I 've consult **pvlib.bifacial.infinite_sheds** but I didn't find function that really can calculate view factor for rear side all functions are about sky view factor
any solution please ! many thanks in advance

**Comment by cwhanse** [comment]
View factor from the rear side to which surface, the ground or the sky? Or are you looking for something else?

**Reply by 115522** [reply]
view factor from ground to rear side

**Reply by cwhanse** [reply, marked as answer]
The function `pvlib.bifacial.infinite_sheds._vf_row_ground` returns the view factor from a point on the module to the ground, and `_vf_row_ground_integ` integrates those view factors over the shaded and unshaded portions of the module. To specify front or rear, use the `surface_tilt`; for example, `surface_tilt=30` will return view factors for the surface facing toward the sky at 30 tilt, `surface_tilt=150` will return view factors for the side facing the ground.

These view factors are the fraction of radiation leaving the ground that arrives at the module. I assume that's what you want. Maybe our way to describing them isn't as clear as it could be. In pvlib documentation, we say "from the module to the ground" is as if one is viewing the ground from the module. I think heat transfer literature would describe this differently, as the fraction of radiation leaving the ground that is received on the module.

**Reply by 115522** [reply]
so when I want  to work with `_vf_row_ground(x, surface_tilt, gcr)`  x  is the hight of module from ground ?? can explain please

**Reply by cwhanse** [reply]
x is a fraction, of the distance from the bottom edge to the top of the module, along the slanted height.  The _infinite_sheds model is simplified by not having a distance between the module and the ground, and thus modules can't "see" underneath adjacent rows.

**Reply by 115522** [reply]
many thanks for your  availibility , support , and fast replying @cwhanse 
what I have understood that there is no distance between module and ground in infinite_sheds model ? another point in test x is set as an array  `x = np.array([0., 0.5, 1.0])  ` my question here how can I specifiy x value another question if I am using a tracking system surface_tilt will  take many values how I will know which one is for rear side and which  one for front side please help !availibility

**Reply by cwhanse** [reply]
In _vf_row_ground, x can be a single value or an array. surface_tilt can be an array. But _this is important_ x and surface_tilt cannot both be arrays. We haven't vectorized these functions in that manner (yet).

To get a view factor for a length of the module, rather than at a single point, you want to use _vf_row_ground_integ, instead of _vf_row_ground. Use f_x to identify the fraction of the module (from the bottom edge). _vf_row_ground_integ can accept both f_x and surface_tilt as arrays as long as they are the same length.

You will have to keep track of the front and rear surfaces in your code. Usually, trackers rotate to face the front surface east in the morning. The surface_tilt for the rear side will be 180 - surface_tilt for the front sice.

**Reply by 115522** [reply]
thank you , to summary : to calculate view factor for rear side we should use `_vf_row_ground_integ` function and to calculate f_x I should use  Pythagoras Theorem then we make an array from zero to f_x value and for surface tilt , if I have a fixed system I put only the tilt of system else if i have a tracking system I  should put surface tilt but then the result will be view factor for shaded and view factor for unshaded which one represents view factor for rear side , thank you another time

**Reply by cwhanse** [reply]
Yes to all except this: " f_x I should use Pythagoras Theorem then we make an array from zero to f_x value" I think you might be confused about f_x. It is the fraction of the "slant" distance, i.e., the distance along the hypotenuse of the right triangle I think you are imagining.

**Reply by 115522** [reply]
okey so slant distance is **hypotenus / horizontal distance(width)** sorry @cwhanse  but really 
until now I have not understood the result returned by function and view factor loss for rear side
